app/models/container_quota/purging.rb

```diff
@@ -0,0 +1,25 @@
+class ContainerQuota < ApplicationRecord
+  module Purging
+    extend ActiveSupport::Concern
+    include PurgingMixin
+
+    module ClassMethods
+      def purge_date
+        ::Settings.container_entities.history.keep_archived_quotas.to_i_with_method.seconds.ago.utc
+      end
+
+      def purge_window_size
+        ::Settings.container_entities.history.purge_window_size
+      end
+
+      def purge_scope(older_than)
+        where(arel_table[:deleted_on].lteq(older_than))
+      end
+
+      def purge_associated_records(ids)
```

Comment:
reminder: adding `purge_associated_records` currently causes us to retrieve all ids that are deleted from container quota.  If the number of records is low, then this is not a problem.

If there are a bunch of records to be deleted, we may want to come up with another way to clear out these associated 2 tables.

Reply:
I expect few ContainerQuota records purged — only when admins delete the whole quota definition — but possibly many child ContainerQuotaItem records to purge (these may be archived on inventory changes, may become comparable to ContainerGroup table).

I'm doing this to avoid leaving orphan items.  Since items should get archived when parent quota gets archived, we could also avoid orphans by giving items a lower lifetime, e.g. 5.month vs 6.month.  But I prefer ensuring DB consistency by explicit code over "it happens to work if tuned correctly" :-)

Reply:
Scratch that, `purge_associated_records` is also needed to avoid leaving orphan ContainerQuotaScopes, no alternative there.